modules/javafx.graphics/src/test/java/test/com/sun/javafx/util/DataURITest.java

```diff
@@ -174,4 +176,31 @@ public void testLongUriToString() {
         assertEquals("data:text/plain;charset=utf-8;base64,SGVsbG9Xb3JsZE...RIZWxsb1dvcmxk", uri.toString());
     }
 
+    @Test
+    public void testPercentEncodedTextIsDecodedAccordingToRFC3986() {
+        // We use URLEncoder here to escape the emoji character using percent-encoding.
+        // When DataURI parses its payload, it automatically converts percent-encoded characters back to octets.
+        String input = URLEncoder.encode("🙂", StandardCharsets.UTF_8);
```

Comment:
would it make sense to try several different strings that include +, \n, \t, data:, charset:, %, empty string, &, _, %zz?

Reply:
Most of these cases should already be covered by existing tests (`testMissingDataSeparatorIsInvalid`, `testParametersListWithoutKeyValuePairsIsInvalid`, `testLeadingOrTrailingWhitespaceIsAcceptable`).

Reply:
minor(?): Is it common practice for the FX code base to include non-ASCII characters?  If not, perhaps encode this instead of using the smiley directly (or use an ASCII character that requires % escaping, like `%`).